lib/srv/termhandlers.go

```diff
@@ -45,6 +48,32 @@ func (t *TermHandlers) HandleExec(ch ssh.Channel, req *ssh.Request, ctx *ServerC
 		return t.SessionRegistry.OpenSession(ch, req, ctx)
 	}
 
+	// If this code is running on a Teleport node and PAM is enabled, then open a
+	// PAM context.
+	var pamContext *pam.PAM
```

Comment:
can we have a default NOP Context always created and called close on? This could simplify the logic around all those places as it will allow you to call `defer` as well?